alpha/veneer/basic/basic.go

```diff
@@ -52,8 +29,7 @@ func (v Veneer) renderFile(ctx context.Context, ref string) (*declcfg.Declarativ
 
 	for _, b := range cfg.Bundles {
 		if !isBundleVeneer(&b) {
-			outb = append(outb, b)
-			continue
+			return nil, fmt.Errorf("bundle veneers must define only a schema and an image; all other fields must not be defined")
 		}
```

Comment:
The original intent of this code was to allow the author to provide a combination of completely-rendered olm.bundle references as well as 'shorthand-ed' ones which contained only the image references.  
If a bundle contained properties, it was presumed to be of the completely-rendered variety and just accumulated (i.e. the author was responsible for the completeness of the rendering). 
If a bundle was without properties, it was presumed to be a shorthand bundle, and we would render it ourselves and accumulate it to the output. 

I don't love the idea of refusing to accept any completely-rendered bundles in the veneer, but there is some question of why any author would choose to mix-and-match in that fashion anyway.

Reply:
My perspective here is to make things as strict and locked down as possible at first so that we can attempt to keep the scope of what we need to support small. We can always open things up to support that use case in the future, but it we support it now, we're signing up for it now and we might not understand the ways that people might use/abuse it.